pkg/gameserverallocations/controller.go

```diff
@@ -175,7 +176,11 @@ func (c *Controller) allocationDeserialization(r *http.Request, namespace string
 	gsa.TypeMeta = metav1.TypeMeta{Kind: gvks[0].Kind, APIVersion: gvks[0].Version}
 
 	mediaTypes := scheme.Codecs.SupportedMediaTypes()
-	info, ok := k8sruntime.SerializerInfoForMediaType(mediaTypes, r.Header.Get("Content-Type"))
+	mt, _, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
+	if err != nil {
+		return gsa, err
```

Comment:
can we errors.Wrapf(....) this error message before passing it back out, so we can capture context?

Outside of that, this looks good to go.